docs/notebooks/delayed-reward.ipynb

```diff
@@ -393,112 +415,266 @@
-                "    \n",
-                "    # Update the agent\n",
-                "    for token, reward, context_row in zip(decisions, rewards, one_hot_encoding):\n",
-                "        agent.select_for_update(token).update(\n",
-                "            np.atleast_2d(context_row), np.atleast_1d(reward)\n",
-                "        )\n",
+                "    for iteration in range(1000):\n",
+                "        # Start of each week, randomly adjust demand\n",
+                "        if iteration % 7 == 0:\n",
+                "            oracle.adjust_demand()\n",
+                "\n",
+                "        # Recompute best price per city\n",
+                "        best_price_per_city = compute_best_price_per_city(oracle)\n",
                 "\n",
-                "    # End of each month, decay the bandit\n",
-                "    agent.decay(one_hot_encoding[0], decay_rate=0.98)"
+                "        # Start of the day, adjust prices\n",
+                "        decisions = agent.pull(one_hot_encoding)\n",
+                "\n",
+                "        # End of the day, collect rewards\n",
+                "        rewards = [token.take_action(location) for location, token in enumerate(decisions)]\n",
+                "        regret.append((best_price_per_city - np.array(rewards)).sum())\n",
+                "        \n",
+                "        # Update the agent\n",
+                "        for token, reward, context_row in zip(decisions, rewards, one_hot_encoding):\n",
+                "            agent.select_for_update(token).update(\n",
+                "                np.atleast_2d(context_row), np.atleast_1d(reward)\n",
+                "            )\n",
+                "\n",
+                "        # End of each month, decay the bandit\n",
+                "        if iteration % 30 == 0:\n",
+                "            agent.decay(one_hot_encoding[0], decay_rate=decay_rate)\n",
+                "\n",
+                "    return regret\n",
+                "\n",
+                "regret = run_decay_simulation(0.98, oracle)\n"
             ]
         },
         {
             "attachments": {},
             "cell_type": "markdown",
             "metadata": {},
             "source": [
-                "In this case, we can see that the agent is able to keep up with the changing demand, but that does not necessarily result in lower cumulative regret. This is because the agent is exploring more, and therefore choosing suboptimal prices more often. When choosing to use `Agent.decay`, we must be careful to treat the `decay_rate` as a hyperparameter and tune it accordingly.\n"
+                "In this case, we can see that the agent is able to keep up with the changing demand, but that does not necessarily result in lower cumulative regret. This is because the agent is exploring more, and therefore choosing suboptimal prices more often. \n"
```

Comment:
In the plot you have now, it actually _does_ result in lower cumulative regret, right? Should change the wording here to reflect that.